CSharp/DevVmPowershell/DevVmPsModules/AddZipFolderModule.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Text;
+using System.Threading.Tasks;
+using Helpers;
+
+namespace DevVmPsModules
+{
+	[Cmdlet(VerbsCommon.Add, "AgentByName")]
+	public class AddZipFolderModule : BaseModule
+	{
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 0,
+			HelpMessage = "Path of the Folder to be Zipped")]
+		public string SourceFolderPath { get; set; }
+
+		[Parameter(
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
+			ValueFromPipeline = true,
+			Position = 0,
+			HelpMessage = "Path of the where the Zip File should be created")]
+		public string DestinationZipFilePath { get; set; }
+
+		protected override void ProcessRecordCode()
+		{
+			//Validate Input arguments
+			ValidateInputArguments();
+
+			IZipHelper zipHelper = new ZipHelper();
+
+			// Zip Designated Folder
+			zipHelper.ZipFolder(SourceFolderPath, DestinationZipFilePath);
+		}
+
+		private void ValidateInputArguments()
+		{
+			if (string.IsNullOrWhiteSpace(SourceFolderPath))
+			{
+				throw new ArgumentNullException(nameof(SourceFolderPath), $"{nameof(SourceFolderPath)} cannot be NULL or Empty.");
+			}
+
+			if (string.IsNullOrWhiteSpace(DestinationZipFilePath))
+			{
+				throw new ArgumentNullException(nameof(DestinationZipFilePath), $"{nameof(DestinationZipFilePath)} cannot be NULL or Empty.");
+			}
+
+			FileInfo fileInfo = new FileInfo(DestinationZipFilePath);
```

Comment:
we should also do a validation if the parent folder of the destination zip file path exists.